cmds/dhclient/dhclient.go

```diff
@@ -247,12 +281,17 @@ func main() {
 			} else {
 				done <- dhclient6(iface, *renewals, timeout, *retry)
 			}
-		}(i)
+			debug("Done dhclient for %v", ifname)
+		}(i.Attrs().Name)
 	}
 
+	go func() {
+		wg.Wait()
```

Comment:
The waitgroup is actually a fairly standard way to solve this problem, I even see it used in places where the number of goroutines is known, since it continues to work should the code change in a way that removes that knowledge ...